src/foam/u2/search/DateSearchView.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.search',
+  name: 'DateSearchView',
+  extends: 'foam.u2.View',
+
+  documentation: `A SearchView for properties of type Date.`,
+
+  exports: [
+    'as data'
+  ],
+
+  properties: [
+    {
+      name: 'property',
+      documentation: `The property that this view is filtering by. Should be of
+          type Date.`,
+      required: true
+    },
+    {
+      class: 'String',
+      name: 'qualifier',
+      documentation: `Lets the user choose a predicate to filter the view by.`,
+      view: {
+        class: 'foam.u2.view.ChoiceView',
+        choices: [
+          ['True', '--'],
+          ['Gt', 'after'],
+          ['Lt', 'before'],
+        ],
+        defaultValue: 'True'
+      }
+    },
+    {
+      class: 'Date',
+      name: 'date',
+      documentation: `A date selectable by the user. Used as the second argument
+          to the predicate generated by this view.`,
+    },
+    {
+      name: 'predicate',
+      documentation: `All SearchViews must have a predicate as required by the
+          SearchManager. The SearchManager will read this predicate and use it
+          to filter the dao being displayed in the view.`,
+      expression: function(qualifier, date) {
+        if ( qualifier.length > 0 && date && ! isNaN(date.valueOf()) ) {
```

Comment:
If Gt it should Gt end-of-day. If Lt it should be Lt end-of-day. 
As mentioned in demos, Between would also be good.
Other options might include: 'Yesterday', 'Today', 'Tomorrow', 'This Week', 'This Month', 'Next Month', etc.

Reply:
This is more or less blocked by the big date/time refactoring that @mcarcaso proposed. I could implement this before the refactor, but then I'd need to go back and change it again afterwards.